playwright/fixtures/index.ts

```diff
@@ -35,113 +44,242 @@ export type Fixtures = {
+                localStorage: [],
               },
             ],
           },
-        ],
-      },
-    });
+        });
+      }
+
+      // Create new page object with new context
+      const page = await context.newPage();
+
+      // Logout user by default
+      try {
+        await fs.rm(path.join(os.homedir(), ".prismic"));
+      } catch (error) {
+        // Ignore since it means the user is already logged out
+      }
+
+      // Login user if needed
+      if (loggedIn) {
+        // In CI we define a PRISMIC_URL env variable to fasten the tests
+        let prismicUrl: string | undefined = process.env["PRISMIC_URL"];
+
+        // In local we get the Prismic URL from the browser, it helps to avoid
+        // switching manually between Wroom and Prismic
+        if (!prismicUrl) {
+          await page.route(
+            "*/**/_manager",
+            async (route) => {
+              const postDataBuffer = route.request().postDataBuffer() as Buffer;
+              const postData = decode(postDataBuffer) as Record<
+                "procedurePath",
+                unknown[]
+              >;
+
+              if (postData.procedurePath[0] === "getState") {
+                const response = await route.fetch();
+                const existingBody = await response.body();
+                const existingData = (
+                  decode(existingBody) as Record<"data", unknown>
+                ).data as Record<
+                  "env",
+                  {
+                    endpoints: { PrismicWroom: string };
+                  }
+                >;
+
+                // Get Prismic URL from the response of the getState call
+                prismicUrl = existingData.env.endpoints.PrismicWroom;
 
-    const page = await context.newPage();
+                await route.continue();
+              }
+            },
+            // Ensure only the first getState call is intercepted
+            {
+              times: 1,
+            },
+          );
 
-    // Use the fixture value in the test.
-    await use(page);
+          // Visit the page to trigger the getState call and wait for it
+          await page.goto("/");
+          await page.waitForResponse("*/**/_manager");
+        }
 
-    // Gracefully close up everything
-    await page.close();
-    await context.close();
-  },
-});
+        const activeEnv = prismicUrl?.includes("wroom") ? "WROOM" : "PRISMIC";
+        const email = process.env[`${activeEnv}_EMAIL`];
+        const password = process.env[`${activeEnv}_PASSWORD`];
+
+        if (!prismicUrl) {
+          console.warn("Could not find Prismic URL.");
+        } else if (!email || !password) {
+          console.warn(
+            `Missing EMAIL or PASSWORD environment variables for ${activeEnv} environment.`,
+          );
+        } else {
+          // Do the authentication call
+          const res = await fetch(
+            new URL("./authentication/signin", prismicUrl).toString(),
```

Comment:
Do we call the actual API? Is there a way to mock this, or do you prefer calling the real API?

To be honest, I don't have much experiencing writing E2E tests (I usually stick to integration where we would normally mock everything with like-real responses), so I might have the wrong perspective.

Reply:
I prefer the real API. As I said fully e2e, it's important to mock as less as possible. Still being able to mock is cool for standalone test.

But for scenario, as you said with the onboarding we will not setup anything. I agree that integration tests we would mock everything.